### PR TITLE
chore: Reduce complex selector usage in Radio styles

### LINKED ISSUE #25483: Checkbox and Radio styles nested selector usage might not be scalable
Currently the `Checkbox` and `Radio` components use nested selectors to style states, some examples below:

```ts
[`:not(:checked) ~ .${radioClassNames.indicator} > *`]
':enabled'
':enabled:not(:checked)'
```

These kinds of styles have two important consequences that will impact partners when using these components

### Override specificity

It is not obvious how to override slot styles without using `important`: https://codesandbox.io/s/funny-monad-xhjzj1?file=/example.tsx

```ts
const useStyles = makeStyles({
  checkedIndicator: {
    backgroundColor: "red"
  },
  checkedIndicatorImportant: {
    backgroundColor: "red !important"
  },
  checkedNestedIndicator: {
    ":enabled:checked:not(:indeterminate)": {
      [`& ~ .${checkboxClassNames.indicator}`]: {
        backgroundColor: "red"
      }
    }
  }
});
```

Since our components are liable to be used in our libraries that are shared and reused, these kinds of overrides are not scaleable in an app or cross app.

### CSS classes bloat

Each control will contain more >100 CSS classes at any time. The use of nested selectors means that the classes are never applied conditionally and will always be there. This has a negative impact on performance since atomic css performance can decrease linearly with respect to the number of classes.

Checkbox

image

Radio

image

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Radio used complex CSS selectors to directly set colors on the `indicator` and `label` slots. This makes it very difficult to override the colors in custom styles.

## New Behavior

Use CSS variables in the slot styles to set the colors. Any rule that uses a selector (e.g. `:hover`, `:active`, or `:checked`) _only_ sets CSS variables. The slot's base styles reference the CSS variable in zero-specificity rules. This allows users to override slot colors using any rules they choose.

Note: Because the HTML `<input type="radio">` element does not fire `onChange` events when the radio is deselected, it isn't possible to set checked styles via JavaScript. Therefore it's still necessary to use CSS for the checked state, along with a sibling combinator to set the CSS variables on the appropriate slot (`:checked ~ .fui-Radio__indicator`). However, since these complex selectors are only setting CSS variables, it is possible for a user to override the styles directly on the slot without needing a complex selector.
 
## Related Issue(s)

- Fixes #25483

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Avatar-mount_base.html">649</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Avatar-mount.html">647</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Button-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Button-mount.html">291</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Field-mount_base.html">1130</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Field-mount.html">1124</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProvider-mount_base.html">712</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProvider-mount.html">722</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-mount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-mount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">66</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">67</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/MakeStyles-mount_base.html">857</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/MakeStyles-mount.html">862</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Persona-mount_base.html">1742</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/Persona-mount.html">1686</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/SpinButton-mount_base.html">1375</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30246/perf-test-react-components/SpinButton-mount.html">1379</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                    Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                               |  `32.042 kB`<br />`10.046 kB` |  `31.241 kB`<br />`10.206 kB` | `-801 B` <br />`160 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `156.347 kB`<br />`43.735 kB` | `155.546 kB`<br />`43.788 kB` |  `-801 B` <br />`53 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         |     `129.567 kB`<br />`35 kB` | `128.766 kB`<br />`35.075 kB` |  `-801 B` <br />`75 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |  `74.744 kB`<br />`20.215 kB` |  `73.943 kB`<br />`20.295 kB` |  `-801 B` <br />`80 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |  `73.351 kB`<br />`19.806 kB` |   `72.55 kB`<br />`19.893 kB` |  `-801 B` <br />`87 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.901 kB`<br />`20.261 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `209.786 kB`<br />`59.864 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `42.388 kB`<br />`14.103 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `7.099 kB`<br />`2.385 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `15.345 kB`<br />`6.263 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.899 kB`<br />`13.781 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3b10c5e8637ad42b0943ab8e1f0a47804483d3b5'>3b10c5e8637ad42b0943ab8e1f0a47804483d3b5</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-radio-f2a00719-e7f8-45c1-a97b-76b8505bf4f4.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-radio-f2a00719-e7f8-45c1-a97b-76b8505bf4f4.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30246&commitId=bdbf39e31bfb198dab37eea752110c3686edfdd6&env=prod&clientType=FLUENTUI)


  <details><summary><b>Radio Converged</b> 10 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Radio Converged.disabled - Dark Mode.hover.chromium.png|3| Changed|
|Radio Converged.disabled.active.chromium.png|5| Changed|
|Radio Converged.disabled+checked.default.chromium.png|5| Changed|
|Radio Converged.disabled.hover.chromium.png|5| Changed|
|Radio Converged.disabled - Dark Mode.active.chromium.png|3| Changed|
|Radio Converged.disabled.rest.chromium.png|5| Changed|
|Radio Converged.disabled - Dark Mode.rest.chromium.png|3| Changed|
|Radio Converged.disabled+checked - Dark Mode.default.chromium.png|3| Changed|
|Radio Converged.group-disabled - RTL.default.chromium.png|16| Changed|
|Radio Converged.group-disabled.default.chromium.png|16| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (behowell) [reply change/@fluentui-react-radio-f2a00719-e7f8-45c1-a97b-76b8505bf4f4.json:1]: The screenshot diffs are a result of correcting the disabled border color for the radio indicator. The changed visuals now match the figma ✅

**Comment by layershifter** [comment]
Closing for housekeeping 🏡 Feel free to re-open

**Pull request closed without merging**